### PR TITLE
Improve ReflectometrySumInQ unit tests

### LINKED ISSUE #22723: ReflectometrySumInQ needs unit tests for negative 2theta
The quick fix introduced in #22720 showed that `ReflectometrySumInQ` should be tested more rigorously. Unit test(s) should be added to ensure that the algorithm handles negative 2theta angles flawlessly.

### PR DESCRIPTION
**Pull request by soininen**
This PR adds a unit test for `ReflectometrySumInQ` to cover a bug which has been fixed in PR #22720.

**Report to:** nobody.

**To test:**

Code review.

Fixes #22723.

Does this update require release notes?
- [ ] Yes
- [x] No

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/test/ReflectometrySumInQTest.h [hunk 1]
```diff
@@ -243,7 +273,9 @@ class ReflectometrySumInQTest : public CxxTest::TestSuite {
   }
 
 private:
-  static Mantid::API::MatrixWorkspace_sptr testWorkspace() {
+  static Mantid::API::MatrixWorkspace_sptr
+  testWorkspace(const double centreTwoThetaDegrees = 0.87,
```
Comment (peterfpeterson) [inline Framework/Algorithms/test/ReflectometrySumInQTest.h]: This constant has a bad name. Either it is in radians or your conversion on L291 is wrong.

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
The new test is welcome, but doesn't prevent things from breaking since the result is not checked.

File: Framework/Algorithms/test/ReflectometrySumInQTest.h (hunk 1 above)
Comment (soininen) [reply Framework/Algorithms/test/ReflectometrySumInQTest.h]: I am sorry but I completely fail to see what is wrong with the conversion on L291. Could you elaborate?

Comment (peterfpeterson) [reply Framework/Algorithms/test/ReflectometrySumInQTest.h]: Line 291:
```C++
const double twoTheta{centreTwoThetaDegrees / 180. * M_PI};
```
suggests that `centreTwoThetaDegrees` is actually in radians.

Comment (soininen) [reply Framework/Algorithms/test/ReflectometrySumInQTest.h]: Conversion from radians to degrees would be `twoTheta / M_PI * 180.`, wouldn't it? Am I missing something here?

Comment (peterfpeterson) [reply Framework/Algorithms/test/ReflectometrySumInQTest.h]: `twoTheta` is in degrees. `centreTwoThetaDegrees` is in radians. The name, `centreTwoThetaDegrees` suggests a value that is in degrees.

**Review by peterfpeterson — APPROVED** [review]

File: Framework/Geometry/inc/MantidGeometry/Crystal/AngleUnits.h [hunk 2]
```diff
@@ -33,10 +33,10 @@ namespace Geometry {
  */
 
 /// Degrees to radians conversion factor
-const double deg2rad = M_PI / 180.;
+constexpr double deg2rad = M_PI / 180.;
```
Comment (martyngigg) [inline Framework/Geometry/inc/MantidGeometry/Crystal/AngleUnits.h]: :+1:

**Pull request merged**